File: src/Ticketing.Application/Catalog/Dtos/EventDetailDto.cs
using Ticketing.Domain.Catalog;

namespace Ticketing.Application.Catalog.Dtos;

/// <summary>
/// 活動詳情。<c>MaxTicketsPerBuyer</c> 與 <c>AllowsContiguousAllocation</c> 來自
/// <c>PurchasePolicy</c>——**上限不存資料表**，由 Domain 依類別決定（設計文件 04 第 2.2 節）。
/// </summary>
public sealed record EventDetailDto(
    int Id,
    string Code,
    EventCategory Category,
    string Title,
    string Performer,
    string Genre,
    string Description,
    string ImagePath,
    PerformanceSummaryDto Performance,
    IReadOnlyList<SectionDto> Sections,
    int MaxTicketsPerBuyer,
    bool AllowsContiguousAllocation,
    string Currency,
    DateTimeOffset ServerNowUtc);

File: web/src/features/catalog/model/EventDetailDto.ts
import type { EventCategory } from './EventCategory';
import type { SalesStatus } from './SalesStatus';
import type { SectionDto } from './SectionDto';

export interface EventDetailDto {
  id: number;
  code: string;
  category: EventCategory;
  title: string;
  performer: string;
  genre: string;
  description: string;
  imagePath: string;
  performance: {
    id: number;
    city: string;
    venue: string;
    startsAtUtc: string;
    salesOpensAtUtc: string;
    salesClosesAtUtc: string;
    durationMinutes: number;
    isSalesPaused: boolean;
    salesStatus: SalesStatus;
  };
  sections: SectionDto[];
  maxTicketsPerBuyer: number;
  allowsContiguousAllocation: boolean;
  currency: string;
  serverNowUtc: string;
}
